ktor-http/common/src/io/ktor/http/URLBuilder.kt

```diff
@@ -22,7 +22,7 @@ public const val DEFAULT_PORT: Int = 0
  * @property fragment URL fragment (anchor name)
  * @property trailingQuery keep a trailing question character even if there are no query parameters
  */
-public class URLBuilder(
+public data class URLBuilder(
```

Comment:
This sounds good at first but notice that it makes it difficult to keep binary compatibility because of the `copy` function with a lot of parameters in the case when we need to add a property. Also, adding a property in the middle breaks components functions as they are just ordered by declaration. 
What was the intention of this? Is it because of missing `toString` or because of `clone` vs `copy`?

Reply:
The intention is the problem with the default values. With the default values set inside the constructor and not inside the `build()` function, it is impossible to know, if the user wants to request the relative `/some` or the explicit url `localhost/some`, because both url strings creates the same url builder. This pr use generated `toEquals` function for the missing equality to compare two instances. Of course this is also possible with a custom `toEquals` function instead.